apis/v1alpha3/servicebinding_webhook.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var servicebindinglog = logf.Log.WithName("servicebinding-resource")
+
+func (r *ServiceBinding) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/mutate-servicebinding-io-v1alpha3-servicebinding,mutating=true,failurePolicy=fail,sideEffects=None,groups=servicebinding.io,resources=servicebindings,verbs=create;update,versions=v1alpha3,name=mservicebinding.kb.io,admissionReviewVersions={v1,v1beta1}
+
+var _ webhook.Defaulter = &ServiceBinding{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *ServiceBinding) Default() {
+	servicebindinglog.Info("default", "namespace", r.Namespace, "name", r.Name)
+
+	if r.Spec.Name == "" {
+		r.Spec.Name = r.Name
+	}
```

Comment:
I'm fine dropping the mutating web hook registration, but I think we should keep the default method. The defaults can be applied inside the reconciler instead of being persisted.

I'm 👎 on introducing a getter method. It will create more confusion than benefit because every other property is treated as a field.

Reply:
> The defaults can be applied inside the reconciler instead of being persisted.

Agree, that can work as well.